FHIRServicesExternalModule.php

```diff
@@ -2805,19 +2807,45 @@ function sendToRemoteFHIRServer($resource){
             throw new \Exception($message);
         };
 
+        $successful_response_with_empty_body = false;
+        if(empty($response)){
+            // parse for response HTTP status code
+            $http_response_line = $http_response_header[0];
+            // look for "HTTP/1.1 [[STATUS_CODE]] OK"
+            if(preg_match("/\s([0-9]+)\s/", $http_response_line, $match)){
+                $http_response_status_code = intval($match[0]);
+                if($http_response_status_code>=200 && $http_response_status_code<300){
+                    $successful_response_with_empty_body = true;
```

Comment:
Would it be simpler but just as effective to return `null` here, and revert the changes below this line?  I don't think there's any reason to continue on to the parsing & type check if we already know the response is empty.

Reply:
Yes, you are right.
My initial intention was to keep the dataflow equivalent for all cases and also let empty requests be processed through the `$this->parse($response);`-logic.
But we can also take a shortcut here and save multiple LoCs.